### DISCUSSION TITLE
File type, textmate and syntax coloration?

### DISCUSSION THREAD

**Discussion by onriv** [redhat-developer/lsp4ij#425] [Q&A] [answered]
Hi in the documentation I saw that for using textmate bundle coloration it’s better not defining a file type:

>[File name pattern mapping](https://github.com/redhat-developer/lsp4ij/discussions/new?category=q-a#file-name-pattern-mapping) with the com.redhat.devtools.lsp4ij.fileNamePatternMapping extension point used to associate a simple pattern file name (ex: *.less) with a language server. This mapping can be very helpful if you need to support syntax coloration with TextMate. Indeed, when you manually define a [File type](https://www.jetbrains.com/help/idea/creating-and-registering-file-types.html) for some file name patterns, you loose the TextMate syntax coloration.

Can we use the textmate syntax coloration and lsp semantic coloration both? The semantic coloration is better in viewing but sometimes slow. It’s nice if it’s possible first coloring with textmate bundle and after the lsp return semantic tokens and recoloring it again .

**Comment by onriv** [comment, marked as answer]
I tested that with file name pattern mapping textmate bundle and lsp colorization can work together. Using file type mapping the textmate bundle indeed doesn’t work

**Reply by angelozerr** [reply]
It is a problem from IntelliJ IDEA. You cannot using TextMate and custom file type both.

When TextMate is associated to a file, the language of this file is textmate. If you associate a file type to your file, your file have not any more the textmate language.

**Reply by onriv** [reply]
Yeah indeed it’s the case. Is there a way add fallback for lsp coloration yet still with a file type? I am wondering if there’s something lost like build/run without a filetype. I found https://youtrack.jetbrains.com/issue/KTIJ-18678/Implement-instantaneous-non-semantic-syntax-highlighting and https://github.com/antlr/intellij-plugin-v4 but no clue for if it is possible

**Reply by angelozerr** [reply]
If your question is I want to use textmate and use file type, the answer is no it is not possible.

I suggest that you ask about using textmate znd file type both to jetbrains

**Reply by onriv** [reply]
Yeah indeed thanks for the suggestion!
